### PR TITLE
fix(HMS-1847):Aws_cloudwash_pipeline

### PR DESCRIPTION
**Pull request by mayurilahane**
parameterizing pod template to use Azure as well AWS for the cleanup.

### CODE REVIEW

File: deploy/cloudwash_job_aws.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            restartPolicy: Never
+            containers:
+              - name: provisioning-cleanup
+                image: quay.io/redhatqe/cloudwash
+                imagePullPolicy: Always
+                command:
+                  - swach
+                args:
+                  - "-d"
+                  - "aws"
```
[the remark this answers is not in the document]
Comment (mayurilahane) [reply deploy/cloudwash_job_aws.yaml:32]: and what about env var?
it varies for every hypervisor
aws needs 2 var
azure needs 6 var

File: deploy/cloudwash_job.yaml [hunk 2]
```diff
@@ -66,6 +64,16 @@ objects:
                       secretKeyRef:
                         key: location
                         name: provisioning-azure-auth-secret
+                  - name: CLEANUP_AWS__AUTH__ACCESS_KEY # AWS creds
+                    valueFrom:
+                      secretKeyRef:
+                        key: access_key_id
+                        name: cloudwash-secret
```
Comment (mshriver) [inline deploy/cloudwash_job.yaml:71]: Do you have an app-interface MR for the namespace to include this secret?

Comment (mayurilahane) [reply deploy/cloudwash_job.yaml:71]: yep here
https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/89921/diffs

**Comment by mayurilahane** [comment]
@mshriver or @akhil-jha wanna give it a go ?

File: deploy/cloudwash_job.yaml [hunk 3]
```diff
@@ -73,3 +80,6 @@ parameters:
   description: "Schedule for cron job"
   value: ''
   required: true
+- name: CLOUDWASH_ARGS
+  required: true
+  value: '["aws", "azure", "--nics", "--vms", "--discs", "--images"]'
```
Comment (akhil-jha) [inline deploy/cloudwash_job.yaml:85]: We don't want to delete images.

Comment (akhil-jha) [reply deploy/cloudwash_job.yaml:85]: `'["aws", "azure", "--nics", "--vms", "--discs", "--images"]' ` is not a valid command. 
Can you confirm it?

**Review by akhil-jha — COMMENTED** [review]
I would say having a working azure cleanup takes higher priority than add all the clouds for now.

File: deploy/cloudwash_job.yaml (hunk 3 above)
Comment (mayurilahane) [reply deploy/cloudwash_job.yaml:85]: Thanks for the review!
I would like to get some feedback from @mshriver too.
and yes those are parameterized lists for both the cloud resources.
that's why it's like combined params but yeah let's discuss it when mike is back.

Comment (mayurilahane) [reply deploy/cloudwash_job.yaml:85]: looks like cloudwash will only support one cloud at a time.
and as per akhil mentioned i am creating a new container for other resource 
Thanks !

Comment (mayurilahane) [reply deploy/cloudwash_job.yaml:85]: ref : https://redhat-internal.slack.com/archives/C04B2FX09FV/p1705412248915979?thread_ts=1705355987.988969&cid=C04B2FX09FV

Comment (akhil-jha) [reply deploy/cloudwash_job.yaml]: Please don't post the internal links on github

File: deploy/cloudwash_job.yaml [hunk 4]
```diff
@@ -66,6 +67,30 @@ objects:
                       secretKeyRef:
                         key: location
                         name: provisioning-azure-auth-secret
+            containers:
```
Comment (mshriver) [inline deploy/cloudwash_job.yaml:70]: This line is breaking the spec - the containers map is already defined on L24, this should be an additional item in that map.

File: deploy/cloudwash_job.yaml [hunk 5]
```diff
@@ -56,16 +57,30 @@ objects:
...
-                  - name: CLEANUP_AZURE__AUTH__RESOURCE_GROUPS
+              - name: provisioning-aws-cleanup
+                image: quay.io/redhatqe/cloudwash
+                imagePullPolicy: Always
+                command:
+                  - swach
+                args:
+                  - "-d" # Need confirmation from jakub or akhil then only we can remove this flag.
+                  - "aws"
+                  - "--all"
```
Comment (akhil-jha) [inline deploy/cloudwash_job.yaml:68]: Only vms for aws and gcp. 
The structure of azure vs aws/gcp is different on how we are billed.

Comment (akhil-jha) [reply deploy/cloudwash_job.yaml:68]: and Yes, we can remove the `-d` parameter.

Comment (akhil-jha) [reply deploy/cloudwash_job.yaml:68]: We don't care about the vm prefix for these clouds.

Comment (mayurilahane) [reply deploy/cloudwash_job.yaml:68]: added only vms flag and removed -d flag 
thanks!

File: deploy/cloudwash_job.yaml [hunk 6]
```diff
@@ -57,6 +57,30 @@ objects:
                       secretKeyRef:
                         key: subscription_id
                         name: provisioning-azure-auth-secret
+              - name: provisioning-aws-cleanup
+                image: quay.io/redhatqe/cloudwash
+                imagePullPolicy: Always
+                command:
+                  - swach
+                args:
+                  - "-d"
```
Comment (mshriver) [inline deploy/cloudwash_job.yaml:66]: thanks for starting with dryrun here

**Review by mshriver — APPROVED** [review]

**Review by mshriver — APPROVED** [review]

**Pull request merged**